src/main/java/org/primefaces/util/DynamicContentSrcBuilder.java

```diff
@@ -65,36 +63,43 @@ public static String build(FacesContext context, Object value, UIComponent compo
         }
         else if (value instanceof String) {
             src = ResourceUtils.getResourceURL(context, (String) value);
+            return encodeResourceURL(context, src, cache);
         }
         else if (value instanceof StreamedContent) {
             StreamedContent streamedContent = (StreamedContent) value;
-
-            if (stream) {
-                Resource resource = context.getApplication().getResourceHandler().createResource(
-                        "dynamiccontent.properties", "primefaces", streamedContent.getContentType());
-                String resourcePath = resource.getRequestPath();
-
-                Map<String, Object> session = context.getExternalContext().getSessionMap();
-                Map<String, String> dynamicResourcesMapping = (Map) session.get(Constants.DYNAMIC_RESOURCES_MAPPING);
-                if (dynamicResourcesMapping == null) {
-                    dynamicResourcesMapping = new LimitedSizeHashMap<>(200);
-                    session.put(Constants.DYNAMIC_RESOURCES_MAPPING, dynamicResourcesMapping);
-                }
-
-                ValueExpression expression = ValueExpressionAnalyzer.getExpression(
+            ValueExpression expression = ValueExpressionAnalyzer.getExpression(
                         context.getELContext(), component.getValueExpression(attributeName));
+            return build(context, streamedContent, component, cache, type, stream, expression);
+        }
+        return encodeResourceURL(context, src, cache);
```

Comment:
@tandraschko are you referring to this line? If so, src could be null originally:

https://github.com/primefaces/primefaces/blob/d05387b93bd5fc68d825bc5d325cc49472110cf5/src/main/java/org/primefaces/util/DynamicContentSrcBuilder.java#L127

Shall I change that into throwing a FacesException?